File: code/back-office/server/src/main/java/ceni/system/votesync/model/entity/alert/Alert.java
package ceni.system.votesync.model.entity.alert;

import java.sql.Date;

import ceni.system.votesync.model.base.entity.AbstractEntity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Table;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = false)
@Entity
@Table(name = "alertes_importees")
public class Alert extends AbstractEntity<Integer> {

    @Column(name = "id_type_alerte")
    private Integer alertTypeId;

    @Column(name = "id_election")
    private Integer electionId;

    @Column(name = "code_bv")
    private String pollingStationCode;

    @Column(name = "date_alerte")
    private Date alertDate;

    private String description;

    @Column(name = "etat")
    private Integer status;
}

File: code/back-office/front/src/app/dto/alert.ts
export class Alert {
  id: number;
  alertTypeId: number;
  alertType: string;
  alertLevel: number;
  electionId: number;
  election: string;
  pollingStationId: number;
  pollingStation: string;
  alertDate: string;
  description: string;
  status: number;

  constructor(
    id: number,
    alertTypeId: number,
    alertType: string,
    alertLevel: number,
    electionId: number,
    election: string,
    pollingStationId: number,
    pollingStation: string,
    alertDate: string,
    description: string,
    status: number
  ) {
    this.id = id;
    this.alertTypeId = alertTypeId;
    this.alertType = alertType;
    this.alertLevel = alertLevel;
    this.electionId = electionId;
    this.election = election;
    this.pollingStationId = pollingStationId;
    this.pollingStation = pollingStation;
    this.alertDate = alertDate;
    this.description = description;
    this.status = status;
  }
}
